crates/cargo-codspeed/src/app.rs

```diff
@@ -50,60 +54,38 @@ enum Commands {
     },
     /// Run the previously built benchmarks
     Run {
-        /// Optional list of benchmarks to run (run all found benchmarks by default)
-        benches: Option<Vec<String>>,
-
         #[command(flatten)]
-        package_selection: PackageSelection,
+        filters: Filters,
     },
 }
 
-pub fn get_cargo_config() -> Result<GlobalContext> {
+/// Forwards the `codspeed` cfg flag to the compiler through env variable for all future
+fn set_codspeed_rustflags() {
     let mut rustflags = std::env::var("RUSTFLAGS").unwrap_or_else(|_| "".into());
     rustflags.push_str(" -g --cfg codspeed");
     std::env::set_var("RUSTFLAGS", &rustflags);
-    GlobalContext::default()
 }
```

Comment:
Now that we use cargo with the CLI, we could pass this RUSTFLAGS env variable to the CLI command invocation. It would be way cleaner than what we had before

Reply:
Unfortunately, forwarding rustflags from cargo is a bit limited.  
The best way to do so is actually through the RUSTFLAGS environment variable, else we need to use `cargo rustc` which is not really usable since for example you cannot run `cargo rustc --workspace`.

I'll move these RUSTFLAGS as close to the compilation as possible, but IMO we should keep the env var setting